futures-util/src/future/try_join_all_chunked.rs

```diff
@@ -0,0 +1,140 @@
+//! Definition of the `TryJoinAllChunked` combinator, waiting for all of a list of
+//! futures to finish with either success or error whilst only running at most n at a time.
+
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::fmt;
+use core::future::Future;
+use core::mem;
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+use super::{assert_future, TryFuture, TryMaybeDone};
+
+fn iter_pin_mut<T>(slice: Pin<&mut [T]>) -> impl Iterator<Item = Pin<&mut T>> {
```

Comment:
Can we move this to a util module, it's already used by join_all and try_join_all.